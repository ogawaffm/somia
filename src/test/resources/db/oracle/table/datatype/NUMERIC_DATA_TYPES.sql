BEGIN
    EXECUTE IMMEDIATE 'DROP TABLE TEST.NUMERIC_DATA_TYPES';
EXCEPTION
    WHEN OTHERS THEN
        IF SQLCODE != -942 THEN
            RAISE;
        END IF;
END;
/
CREATE TABLE TEST.NUMERIC_DATA_TYPES
(

    SMALLINT_COL         SMALLINT,
    INT_COL              INT,
    INTEGER_COL          INTEGER,

    REAL_COL             REAL,

    FLOAT_COL            FLOAT,
    FLOAT_1_COL          FLOAT(1),
    FLOAT_24_COL         FLOAT(24),
    FLOAT_25_COL         FLOAT(25),
    FLOAT_53_COL         FLOAT(53),
    FLOAT_126_COL        FLOAT(126),

    DOUBLE_PRECISION_COL DOUBLE PRECISION,

    BINARY_FLOAT_COL     BINARY_FLOAT,
    BINARY_DOUBLE_COL    BINARY_DOUBLE,

    NUMBER_COL           NUMBER,
    DEC_COL              DEC,
    DECIMAL_COL          DECIMAL,
    NUMERIC_COL          NUMERIC,

    NUMBER_1_COL         NUMBER(1),
    DEC_1_COL            DEC(1),
    DECIMAL_1_COL        DECIMAL(1),
    NUMERIC_1_COL        NUMERIC(1),

    NUMBER_1_0_COL       NUMBER(1, 0),
    DEC_1_0_COL          DEC(1, 0),
    DECIMAL_1_0_COL      DECIMAL(1, 0),
    NUMERIC_1_0_COL      NUMERIC(1, 0),

    NUMBER_1_1_COL       NUMBER(1, 1),
    DEC_1_1_COL          DEC(1, 1),
    DECIMAL_1_1_COL      DECIMAL(1, 1),
    NUMERIC_1_1_COL      NUMERIC(1, 1),

    NUMBER_38_COL        NUMBER(38),
    DEC_38_COL           DEC(38),
    DECIMAL_38_COL       DECIMAL(38),
    NUMERIC_38_COL       NUMERIC(38),

    NUMBER_38_0_COL      NUMBER(38, 0),
    DEC_38_0_COL         DEC(38, 0),
    DECIMAL_38_0_COL     DECIMAL(38, 0),
    NUMERIC_38_0_COL     NUMERIC(38, 0),

    NUMBER_38_1_COL      NUMBER(38, 1),
    DEC_38_1_COL         DEC(38, 1),
    DECIMAL_38_1_COL     DECIMAL(38, 1),
    NUMERIC_38_1_COL     NUMERIC(38, 1),

    NUMBER_38_38_COL     NUMBER(38, 38),
    DEC_38_38_COL        DEC(38, 38),
    DECIMAL_38_38_COL    DECIMAL(38, 38),
    NUMERIC_38_38_COL    NUMERIC(38, 38)

)