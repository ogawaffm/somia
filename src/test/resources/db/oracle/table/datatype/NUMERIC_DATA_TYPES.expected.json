{
  "name": "NUMERIC_DATA_TYPES",
  "catalogName": "",
  "schemaName": "TEST",
  "typeName": "TABLE",
  "comment": "",
  "columns": [
    {
      "name": "SMALLINT_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INT_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "INTEGER_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "REAL_COL",
      "type": {
        "name": "FLOAT",
        "typeNumber": 6,
        "className": "java.lang.Double",
        "precision": 63,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "FLOAT_COL",
      "type": {
        "name": "FLOAT",
        "typeNumber": 6,
        "className": "java.lang.Double",
        "precision": 126,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "FLOAT_1_COL",
      "type": {
        "name": "FLOAT",
        "typeNumber": 6,
        "className": "java.lang.Double",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "FLOAT_24_COL",
      "type": {
        "name": "FLOAT",
        "typeNumber": 6,
        "className": "java.lang.Double",
        "precision": 24,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "FLOAT_25_COL",
      "type": {
        "name": "FLOAT",
        "typeNumber": 6,
        "className": "java.lang.Double",
        "precision": 25,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "FLOAT_53_COL",
      "type": {
        "name": "FLOAT",
        "typeNumber": 6,
        "className": "java.lang.Double",
        "precision": 53,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "FLOAT_126_COL",
      "type": {
        "name": "FLOAT",
        "typeNumber": 6,
        "className": "java.lang.Double",
        "precision": 126,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DOUBLE_PRECISION_COL",
      "type": {
        "name": "FLOAT",
        "typeNumber": 6,
        "className": "java.lang.Double",
        "precision": 126,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 24,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "BINARY_FLOAT_COL",
      "type": {
        "name": "BINARY_FLOAT",
        "typeNumber": 100,
        "className": "java.lang.Object",
        "precision": 4,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "BINARY_DOUBLE_COL",
      "type": {
        "name": "BINARY_DOUBLE",
        "typeNumber": 101,
        "className": "java.lang.Object",
        "precision": 8,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMBER_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 0,
        "scale": -127,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMBER_1_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_1_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_1_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_1_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMBER_1_0_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_1_0_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_1_0_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_1_0_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMBER_1_1_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_1_1_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_1_1_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_1_1_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 1,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMBER_38_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_38_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_38_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_38_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMBER_38_0_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_38_0_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_38_0_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_38_0_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMBER_38_1_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_38_1_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_38_1_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_38_1_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMBER_38_38_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 38,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DEC_38_38_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 38,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "DECIMAL_38_38_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 38,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NUMERIC_38_38_COL",
      "type": {
        "name": "NUMBER",
        "typeNumber": 2,
        "className": "java.math.BigDecimal",
        "precision": 38,
        "scale": 38,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 38,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    }
  ],
  "indexes": [
    {
      "name": "",
      "columns": [],
      "isUnique": true,
      "comment": "",
      "qualifierName": "",
      "isClustered": false,
      "isHashed": false,
      "isStatistic": true,
      "whereClause": "",
      "cardinality": 0,
      "pages": 0
    }
  ],
  "collation": "",
  "primaryKey": null,
  "bestRowIdentifier": null
}